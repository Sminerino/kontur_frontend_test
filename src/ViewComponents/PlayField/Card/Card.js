import React from 'react';

class Card extends React.PureComponent {

    onThisCardClick = () => {
        this.props.onCardClick(this.props.index);
    };

    render() {
        if (!this.props.isFlippedUp)
            return (
                <div
                    className='card card_down'
                    onClick={this.onThisCardClick}
                    data-tid='Card'
                >
                    <img
                        src={require('./../../../res/Cards/cardback.png')}
                        className='card__image'
                        alt='Card back'
                    />
                </div>
            );
        if (!this.props.isConfirmed)
            return (
                <div className='card card_up' data-tid='Card-flipped'>
                    <img
                        src={this.props.image}
                        className='card__image'
                        alt='Card'
                    />
                </div>
            );
        return (
            <div className='card card_confirmed'>
                <img
                    src={this.props.image}
                    className='card__image'
                    alt='Card'
                />
            </div>
        );
    }
}

export { Card };