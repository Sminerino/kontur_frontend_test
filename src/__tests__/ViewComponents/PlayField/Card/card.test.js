import { Card } from "../../../../ViewComponents/PlayField/Card/Card";
import React from 'react';
import renderer from 'react-test-renderer';

test('Card component is rendering', ()=> {
    const card = renderer.create(
        <Card
            isFlippedUp={true}
            isConfirmed={true}
            onClick
        />
    );

    let cardJSON = card.toJSON();

    expect(cardJSON).toMatchSnapshot();
});