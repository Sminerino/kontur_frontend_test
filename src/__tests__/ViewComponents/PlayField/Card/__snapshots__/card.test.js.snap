// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component is rendering 1`] = `
<div
  className="card card_confirmed"
>
  <img
    alt="Card"
    className="card__image"
    src={undefined}
  />
</div>
`;
